torch/fx/experimental/sym_node.py

```diff
@@ -755,8 +755,75 @@ def _sympy_rshift(a, b):
     return RShift(a, b)
 
 
+def _binary_search_insert_arg(ordered_args, new_arg):
+    """
+    If new_arg is found in ordered_args None is returned, else the new
+    ordered_args with new_arg inserted
+    """
+    low, high = 0, len(ordered_args) - 1
+    from sympy.core.basic import _args_sortkey as sort_key
+
+    # Fast path when new_arg > ordered_args[-1], which is common.
+    if len(ordered_args) and sort_key(ordered_args[-1]) < sort_key(new_arg):
+        return ordered_args + [new_arg]
+
+    while low <= high:
+        mid = (low + high) // 2
+        if ordered_args[mid] == new_arg:
```

Comment:
Hmm... so you're telling me how comparisons are expensive. Is there a way to only do one "compare" here (returning -1, 0, 1 depending on the situation)? Right now you have to do two. (And I guess you don't need `sort_key` for the equality test here because it's guaranteed to coincide with regular equality?

Reply:
use basic_compare directly